.github/workflows/release.yml

```diff
@@ -0,0 +1,123 @@
+        # Need GH Token to use gh CLI
+        env:
+          GH_REPO: ${{ env.GH_REPO }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+          echo "GH_PR_BRANCH=$(gh api repos/"$GH_REPO"/pulls | jq -r '[.[] | select(.head.label | contains("release/"))] | .[0] | .head.ref')" >> $GITHUB_ENV
+      - name: Ensure that version tag matches branch version
+        env: 
+          GH_PR_BRANCH: ${{ env.GH_PR_BRANCH }}
+        run: |
+          BRANCH_VERSION=$(echo "$GH_PR_BRANCH" | tr -d 'release/' )
+
+          if [ "$BRANCH_VERSION" != "$GIT_TAG" ]
+          then
+            echo "ERROR: Version number from tag does not match the version number from the branch name."
+            exit 1
+          fi
+      - name: Checkout head of branch
+        env:
+          GH_PR_BRANCH: ${{ env.GH_PR_BRANCH }}
+        run: |
+          git pull origin "$GH_PR_BRANCH" --ff-only
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Set release version from tag name
+        env:
+          GIT_TAG: ${{ env.GIT_TAG }}
+        run: echo "STREAMLIT_RELEASE_VERSION=$GIT_TAG" >> $GITHUB_ENV
+      - name: Update version
+        env:
+          STREAMLIT_RELEASE_VERSION: ${{ env.STREAMLIT_RELEASE_VERSION }}
+        run: python scripts/update_version.py "$STREAMLIT_RELEASE_VERSION"
+      - name: Create Package
+        timeout-minutes: 120
+        run: |
+          sudo apt install rsync
+          make package
+      - name: Run CLI regression tests
+        run: make cli-regression-tests
+      - name: Store Package
+        uses: actions/upload-artifact@v3
+        with:
+          name: Release
+          path: lib/dist
+      - name: Initialize .pypirc
+        env:
+          STREAMLIT_PYPI_USERNAME: ${{ secrets.STREAMLIT_PYPI_USERNAME }}
+          STREAMLIT_PYPI_API_TOKEN: ${{ secrets.STREAMLIT_PYPI_API_TOKEN }}
+        run: |
+          cd lib
+          echo -e "[pypi]" >> ~/.pypirc
+          echo -e 'username = "$STREAMLIT_PYPI_USERNAME"' >> ~/.pypirc
+          echo -e 'password = "$STREAMLIT_PYPI_API_TOKEN"' >> ~/.pypirc
+      - name: Upload to PyPI
+        run: |
+          make distribute
+      - name: Commit version updates
+        env:
+          GH_PR_BRANCH: ${{ env.GH_PR_BRANCH }}
+          STREAMLIT_RELEASE_VERSION: ${{ env.STREAMLIT_RELEASE_VERSION }}
+        run: |
+          git config user.email "core+streamlitbot-github@streamlit.io"
+          git config user.name "Streamlit Bot"
+
+          git switch -c "$GH_PR_BRANCH"
+
+          git commit -am 'Up version to "$STREAMLIT_RELEASE_VERSION"' && git push origin "$GH_PR_BRANCH" || echo "No changes to commit"
+      - name: Create GitHub Release
+        env:
+          GIT_TAG: ${{ env.GIT_TAG }}
+        run: |
+          eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+          gh release create "$GIT_TAG" -n "" -t "$GIT_TAG"
```

Comment:
Might also warrant a Python script to directly hit the corresponding GH REST API do to this (https://docs.github.com/en/rest/releases/releases#create-a-release) so that we can avoid needing to install the `gh` CLI tool.

Not entirely sure if it's something to be avoided -- I'm not sure if the tool was meant to be used in scripts. My impression with it is that it was intended to be used by humans using a terminal, though, rather than by automated processes

Reply:
Refactored this step into `create_release.py` script to use gh rest api for release creation. 

FWIW didn't find a clear indication that this was a bad practice and found many instances of others using gh api within python scripts.